### PR TITLE
Enable BSTR Field Marshaller for x-plat

### PR DESCRIPTION
**Pull request by luqunl**
Enable Marshalling BSTR as field in PInvoke for x-plat

### CODE REVIEW

**Comment by luqunl** [comment]
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please

**Comment by luqunl** [comment]
@dotnet-bot test Windows_NT arm Cross Debug Innerloop Build please

**Comment by luqunl** [comment]
@dotnet-bot test Ubuntu16.04 arm64 Cross Checked Innerloop Build and Test please

**Comment by luqunl** [comment]
@jashook, Would you take a look  changes in src/vm/callingconvention.h and src/vm/methodtable.cpp

File: src/vm/callingconvention.h [hunk 1]
```diff
@@ -1016,7 +1016,7 @@ int ArgIteratorTemplate<ARGITERATOR_BASE>::GetNextOffset()
     case ELEMENT_TYPE_VALUETYPE:
     {
 #ifdef UNIX_AMD64_ABI
-        MethodTable *pMT = m_argTypeHandle.AsMethodTable();
+        MethodTable *pMT = m_argTypeHandle.GetMethodTable();
```
Comment (janvorli) [inline src/vm/callingconvention.h]: @luqunl can you please educate me on why was this change needed?

Comment (luqunl) [reply src/vm/callingconvention.h]: For some cases, m_argTypeHandle will point to typepamaterdesc instead of methodtable.

**Comment by janvorli** [comment]
> It seems a little weird to remove the BSTR from the FEATURE_COMINTEROP definitions.

It feels ok to me. I see the BSTR as just another type, not necessarily COM interop specific.

File: src/vm/fieldmarshaler.cpp [hunk 2]
```diff
@@ -2383,14 +2381,14 @@ VOID FieldMarshaler_BSTR::DestroyNativeImpl(LPVOID pNativeValue) const
     
     if (pBSTR)
     {
-        _ASSERTE (GetModuleHandleA("oleaut32.dll") != NULL);
         // BSTR has been created, which means oleaut32 should have been loaded.
```
Comment (AaronRobinsonMSFT) [inline src/vm/fieldmarshaler.cpp]: Please update or remove this comment.

File: tests/src/Interop/StringMarshalling/BSTR/PinvokeDef.cs [hunk 3]
```diff
@@ -9,7 +9,14 @@
 
 namespace NativeDefs
 {
-    
+    public struct Person
+    {
+        public int age;
+        public int _pedding;
```
Comment (AaronRobinsonMSFT) [inline tests/src/Interop/StringMarshalling/BSTR/PinvokeDef.cs]: Believe you means `_padding` :)

**Comment by luqunl** [comment]
> It feels ok to me. I see the BSTR as just another type, not necessarily COM interop specific.

Yes, We just treat BSTR as a normal 'well know' type which works x-plat.

File: src/vm/callingconvention.h (hunk 1 above)
Comment (danmosemsft) [reply src/vm/callingconvention.h]: I saw this comment in typehandle.h
```
    // Unlike AsMethodTable, GetMethodTable will get the method table
    // of the type, regardless of whether it is an array etc. Note, however
    // this method table may be shared, and some types (like TypeByRef), have
    // no method table (and this function returns NULL for them)
    inline PTR_MethodTable GetMethodTable() const;
```
